File: Applications/Cpp/Trees/BinarySearchTrees/RangeSumBST.c
/**
 * Definition for a binary tree node.

 */

struct TreeNode
{
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

int rangeSumBST(struct TreeNode *root, int low, int high)
{
    if (root == 0)
    {
        return 0;
    }

    int sum = 0;
    if (root->val > low)
    {
        sum += rangeSumBST(root->left, low, high);
    }
    if (low <= root->val && root->val <= high)
    {
        sum += root->val;
    }
    if (root->val < high)
    {
        sum += rangeSumBST(root->right, low, high);
    }

    return sum;
}

int main(void)
{
    return 0;
}

File: Applications/CSharp/Trees/RangeSumBST.cs
class RangeSumBST
{
    class TreeNode
    {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode(int val = 0, TreeNode left = null, TreeNode right = null)
        {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    };

    static int RangeSum(TreeNode root, int low, int high)
    {
        if (root == null)
            return 0;

        int sum = 0;
        if (root.val > low)
        {
            sum += RangeSum(root.left, low, high);
        }

        if (low <= root.val && root.val <= high)
        {
            sum += root.val;
        }

        if (root.val < high)
        {
            sum += RangeSumBST(root.right, low, high);
        }
        return sum;
    }
}



